src/java.base/share/classes/java/net/Socket.java

```diff
@@ -571,6 +569,9 @@ void setConnected() {
     /**
      * Connects this socket to the server.
      *
+     * <p> If the connection cannot be established, then the socket is closed
+     * and an {@link IOException} is thrown.
```

Comment:
What happened to "the endpoint is an unresolved InetSocketAddress"? If part of the change is to close when called with an unresolved address then it needs to be part of the specification.

Reply:
Fixed in 0761b36297679c9cdea65191f34189b3f51169b0.